state/entrySyncing.go

```diff
@@ -182,11 +182,12 @@ func (s *State) GoSyncEntries() {
 
 	for {
 
+		highestSavedBlk := s.GetHighestSavedBlk()
```

Comment:
Besides that GetHighestSavedBlk is more work than retrieving a variable it doesn't seem like a good idea that this counter changes during the loop iteration.